source/extensions/transport_sockets/tls/cert_validator/default_validator.cc

```diff
@@ -457,9 +458,10 @@ void DefaultCertValidator::addClientValidationContext(SSL_CTX* ctx, bool require
     // Check for duplicates.
     // Note that BoringSSL call only returns 0 or 1.
     // OpenSSL can also return -1, for example on sk_find calls in an empty list
-    if (sk_X509_NAME_find(list.get(), nullptr, name) == 1) {
```

Comment:
do we really need this change for ppc64le? If so, out of curiousity, why? And can we make them specific to that platform?

Reply:
sk_X509_NAME_find definition in openssl and BoringSSL conflict , hence commented

Reply:
Could you elaborate for a bit? I would like to understand what issue gets resolved by commenting this, and also if that represents a general improvement or something that should be specific to ppc64le.

While I write that, future readers of this code may have similar questions. For example it's going to be helpful for those merging newer upstream commits into this to document the rationale for any changes as a hint for how to best handle conflict resolution. So instead of explaining it here, I would like to request code-level comments (the current code-level comment about the openssl/boringssl divergence was also requested in review, and imho it's helpful right now)

Also -- while looking at these changes again, I notice that there's inconsistent return value checking in the original code for the default & spiffe validator when it comes to `sk_X509_NAME_find`. Did you notice that as well? If not, does only one of these need commenting to achieve your goal, or both?

Reply:
if (sk_X509_NAME_find(list.get(), nullptr, name) == 1) {
        if (sk_X509_NAME_find(list.get(), name) == 1) {
                continue;
         }
        }
sk_X509_NAME_find  function definition  required 3 parameter in boringssl while openssl required only 2 params.  In current code build  it refer to boringssl and give error for "sk_X509_NAME_find(list.get(), name) == 1". 

we don't have clarity why this double check done first with 3 params  and then with 2 params. 
for PPC64le we suggest to have modify above lines as below please let me know your view on this. 
 if (sk_X509_NAME_find(list.get(), nullptr, name) == 1) {
                        continue;
         }

Reply:
I'm sorry, I think I misunderstand.. Isn't your suggestion on par with what the original code looks like[1]?

As far as I can tell, the double check was introduced in this PR, and isn't there in the current status of the maistra-2.2 branch. 

[1] https://github.com/maistra/envoy/blob/f44002d75fbc56faadbd3863da39f0094f0ce0a9/source/extensions/transport_sockets/tls/cert_validator/default_validator.cc#L460-L462